### PR TITLE
Adding Stats and effects to NPC's

### PR DESCRIPTION
**Pull request by carlosazpurua1004**
- added stat and effect as field to NPC struct
- initialize them as NULL in NPC_init
- removed health and xp fields and associated functions

- changed name of function "add_stat_player" to the more appropriate "add_stat"

### CODE REVIEW

**Comment by carlosazpurua1004** [comment]
Apperantly NPC code I changed wasn't supposed to be in Dev, the actual NPC code is elsewhere, so i'll have to find and change that code.

File: include/game-state/npc.h [hunk 1]
```diff
@@ -11,8 +11,8 @@ typedef struct npc
 {
     /* hh is used for hashtable, as provided in uthash.h */
     UT_hash_handle hh;
-    char *npc_id;
-    int health;
+    stats_t *stats;
```
Comment (ecoble) [inline include/game-state/npc.h:14]: As already noted in the comment, these changes should go in the NPC module

File: src/npc/src/npc.c [hunk 2]
```diff
@@ -1,26 +1,27 @@
...
     npc->inventory = NULL;
+    npc->stats = NULL;
+    npc->effects = NULL;
 
     return SUCCESS;
 }
 
 /* See npc.h */
-npc_t* npc_new(char *npc_id, int health)
+npc_t* npc_new(char *npc_id)
```
Comment (sumedhsarwate) [inline src/npc/src/npc.c:17]: I think the * needs to be moved

File: include/npc/npc.h [hunk 3]
```diff
@@ -3,14 +3,16 @@
...
 //TODO-#include "dialogue.h"
 
 /* A non-playable character in game */
 typedef struct npc {
     /* hh is used for hashtable, as provided in uthash.h */
     UT_hash_handle hh;
     char *npc_id;
-    int health;
+    stats_t *stats;
+    stat_effect_t *effects;
```
Comment (sumedhsarwate) [inline include/npc/npc.h:15]: These are hash tables right? Would it be helpful to change the type to stats_hash_t and effects_hash_t?

File: src/npc/src/npc.c (hunk 2 above)
Comment (carlosazpurua1004) [reply src/npc/src/npc.c:17]: The return one or the parameter?

Comment (sumedhsarwate) [reply src/npc/src/npc.c:17]: the return one

**Comment by carlosazpurua1004** [comment]
Going to make these changes on a different new branch to avoid merge conflicts etc.

**Pull request closed without merging**